src/box/func.c

```diff
@@ -567,6 +568,9 @@ func_access_check(struct func *func)
 	if ((func_access & PRIV_U) != 0 ||
 	    (func->def->uid != credentials->uid &&
 	     func_access & ~func->access[credentials->auth_token].effective)) {
+		if (func->def->language == FUNC_LANGUAGE_LUA)
+			return access_check_lua_call(func->def->name,
+						     func->def->name_len);
```

Comment:
Shouldn't we check, whether the function is persistent?

Reply:
After an offline discussion, agreed that `lua_call` shouldn't work for persistent Lua functions.

<details><summary>diff</summary>

```diff
diff --git a/src/box/func.c b/src/box/func.c
index c52cdac81006..0dfea46254f0 100644
--- a/src/box/func.c
+++ b/src/box/func.c
@@ -568,7 +568,8 @@ func_access_check(struct func *func)
 	if ((func_access & PRIV_U) != 0 ||
 	    (func->def->uid != credentials->uid &&
 	     func_access & ~func->access[credentials->auth_token].effective)) {
-		if (func->def->language == FUNC_LANGUAGE_LUA)
+		if (func->def->language == FUNC_LANGUAGE_LUA &&
+		    func->def->body == NULL)
 			return access_check_lua_call(func->def->name,
 						     func->def->name_len);
 		/* Access violation, report error. */
diff --git a/test/box-luatest/gh_8803_exec_priv_test.lua b/test/box-luatest/gh_8803_exec_priv_test.lua
index 5aa3248650dc..fbb9a6067465 100644
--- a/test/box-luatest/gh_8803_exec_priv_test.lua
+++ b/test/box-luatest/gh_8803_exec_priv_test.lua
@@ -147,16 +147,16 @@ g.after_test('test_lua_call_func', function(cg)
 end)
 
 -- Checks that execute privilege granted on lua_call grants access to
--- Lua functions registered in _func except for built-ins.
+-- Lua functions registered in _func except for built-ins and persistent
+-- functions.
 g.test_lua_call_func = function(cg)
     local c = cg.conn
     local errfmt = "Execute access to function '%s' is denied for user 'test'"
     cg.grant('test', 'execute', 'lua_call')
-    for _, func in ipairs({'c_func', 'dostring'}) do
+    for _, func in ipairs({'c_func', 'stored_lua_func', 'dostring'}) do
         t.assert_error_msg_equals(errfmt:format(func), c.call, c, func)
     end
     t.assert(c:call('lua_func'))
-    t.assert(c:call('stored_lua_func'))
 end
 
 -- Checks that execute privilege granted on lua_call does not grant access
```

</details>